lib/url.c

```diff
@@ -1781,7 +1781,7 @@ static void llist_dtor(void *user, void *element)
 static struct connectdata *allocate_conn(struct Curl_easy *data)
 {
 #ifdef USE_SSL
-#define SSL_EXTRA + 4 * Curl_ssl->sizeof_ssl_backend_data - sizeof(long long)
+#define SSL_EXTRA (4 * Curl_ssl->sizeof_ssl_backend_data - sizeof(long long))
```

Comment:
So the commit message should say something along the lines:

```
url.c: remove superfluous `+` in `SSL_EXTRA`

When one or more TLS backends is compiled in, `allocate_conn()` wants to reserve enough space for four slots of TLS backend-specific data. To that end, the `SSL_EXTRA` macro is defined which simply reports the delta, so that the four slots as well as the `connectdata` structure can be allocated (and deallocated) in one fell swoop.

The previous version of this code had a leading `+` in the definition of `SSL_EXTRA` (inconsistent with the value `0` in case no TLS backend is compiled in), which this commit removes.

This fixes ...
```

The most important part is the "This fixes ..." which I really would love to see filled in. Because the way I see it, the C compilers I know will handle `1 + + 2` as if you wanted to add `+2` to `1`, and return `3`.

Is there maybe a compiler that mishandles this?